examples/example-protocol/src/assets/rust_plugin_test/expected_types.rs

```diff
@@ -261,5 +261,15 @@ pub struct StructWithGenerics<T> {
 #[serde(rename_all = "camelCase")]
 pub struct StructWithOptions {
     #[serde(default, skip_serializing_if = "String::is_empty")]
-    pub potentially_optional_string: String,
+    pub filled_string: String,
+    #[serde(default, skip_serializing_if = "String::is_empty")]
+    pub empty_string: String,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub filled_option_string: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub empty_option_string: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub never_skipped_filled_option_string: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
```

Comment:
Why do these "never skipped" fields still have the `skip` annotations?

Reply:
This is the default behaviour for the Rust struct generator, it always adds this attribute whenever an `Option` is used if it's not there yet. The type defined in the actual protocol doesn't have these attributes.

Are you saying you'd want to change this behaviour, too? It seems like a decent default.

Reply:
I would change them, yes, because otherwise the behavior is different between the situation when we generate a struct versus when we reuse an existing struct. If we skip the serialization, the resulting type could also never be `null`, it would always be `undefined`. So this led me to look at how the tests could pass.... I think we're making an error by using `assertEquals()` in the tests instead of `assertStrictEquals()`. I highly suspect as soon as we change the assertions, you'll see the values that are asserted to be `null` are really `undefined`.